diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index f89a088..6d76653 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -73,7 +73,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
             {"gliden64", "GLideN64"},
             { NULL, NULL },
         },
-        "gliden64"
+        "angrylion"
     },
     {
         CORE_NAME "-43screensize",
@@ -1386,7 +1386,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
             {"hle", "HLE"},
             { NULL, NULL },
         },
-        "hle"
+        "cxd4"
     },
     {
         CORE_NAME "-FrameDuping",
