From 2ae29cff80cedb1d21088310c8501a1ba8e35b95 Mon Sep 17 00:00:00 2001
From: Zoe Gidiere <duplexsys@protonmail.com>
Date: Fri, 17 Oct 2025 13:35:37 -0600
Subject: [PATCH 2/6] Working Defaults

---
 libretro/libretro_core_options.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index f89a088a..6d766532 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -73,7 +73,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
             {"gliden64", "GLideN64"},
             { NULL, NULL },
         },
-        "gliden64"
+        "angrylion"
     },
     {
         CORE_NAME "-43screensize",
@@ -1381,12 +1381,12 @@ struct retro_core_option_v2_definition option_defs_us[] = {
         NULL,
         NULL,
         {
-            {"cxd4", "CXD4"},
             {"parallel", "ParaLLEl"},
+            {"cxd4", "CXD4"},
             {"hle", "HLE"},
             { NULL, NULL },
         },
-        "hle"
+        "parallel"
     },
     {
         CORE_NAME "-FrameDuping",
-- 
2.53.0

