From c7bf17de860187007df99b9a6ad60fc80dfe3785 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Wed, 3 Sep 2025 00:02:30 -0700
Subject: [PATCH 2/4] Default to angrylion

---
 libretro/libretro_core_options.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index f89a088ad..78492432d 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -73,7 +73,7 @@ struct retro_core_option_v2_definition option_defs_us[] = {
             {"gliden64", "GLideN64"},
             { NULL, NULL },
         },
-        "gliden64"
+        "angrylion"
     },
     {
         CORE_NAME "-43screensize",
-- 
2.43.0

