From 78234d7a8269fa9c66647b6a1f7b9d742a5f620a Mon Sep 17 00:00:00 2001
From: Zoe Gidiere <duplexsys@protonmail.com>
Date: Fri, 17 Oct 2025 13:35:37 -0600
Subject: [PATCH 2/7] Default to ParaLLEl-RSP

---
 libretro/libretro_core_options.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libretro/libretro_core_options.h b/libretro/libretro_core_options.h
index f89a088a..73930937 100644
--- a/libretro/libretro_core_options.h
+++ b/libretro/libretro_core_options.h
@@ -1381,12 +1381,12 @@ struct retro_core_option_v2_definition option_defs_us[] = {
         NULL,
         NULL,
         {
-            {"cxd4", "CXD4"},
             {"parallel", "ParaLLEl"},
+            {"cxd4", "CXD4"},
             {"hle", "HLE"},
             { NULL, NULL },
         },
-        "hle"
+        "parallel"
     },
     {
         CORE_NAME "-FrameDuping",
-- 
2.50.1 (Apple Git-155)

