From 1e066c7cc94db2ae6932078d78bdd3ef7eb4f25e Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 3 Mar 2026 19:35:25 -0800
Subject: [PATCH 7/7] Wait for init before save/load state

---
 libretro/libretro.c | 30 ++++++++++++++++++++++++++----
 1 file changed, 26 insertions(+), 4 deletions(-)

diff --git a/libretro/libretro.c b/libretro/libretro.c
index 86e336f1..7d0d770e 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -25,6 +25,8 @@
 #include <stdlib.h>
 #include <string.h>
 #include <compat/strl.h>
+#include <threads.h>
+#include <time.h>
 
 #include "libretro.h"
 #include "libretro_private.h"
@@ -242,6 +244,14 @@ extern struct
 int pad_pak_types[4];
 int pad_present[4] = {1, 1, 1, 1};
 
+static void sleep_ms(long ms)
+{
+  struct timespec ts;
+  ts.tv_sec  = ms / 1000;
+  ts.tv_nsec = (ms % 1000) * 1000000L;
+  thrd_sleep(&ts, NULL);
+}
+
 static void n64DebugCallback(void* aContext, int aLevel, const char* aMessage)
 {
     char buffer[1024];
@@ -2138,8 +2148,14 @@ size_t retro_serialize_size (void)
 
 bool retro_serialize(void *data, size_t size)
 {
-   if (initializing)
-      return false;
+   // Wait for initialization to complete
+   uint32_t waited_ms = 0;
+   while (initializing)
+   {
+      sleep_ms(1);
+      if (++waited_ms >= 5000)
+        return false;
+   }
 
    retro_savestate_complete = false;
    retro_savestate_result = 0;
@@ -2171,8 +2187,14 @@ bool retro_serialize(void *data, size_t size)
 
 bool retro_unserialize(const void *data, size_t size)
 {
-   if (initializing)
-      return false;
+   // Wait for initialization to complete
+   uint32_t waited_ms = 0;
+   while (initializing)
+   {
+      sleep_ms(1);
+      if (++waited_ms >= 5000)
+        return false;
+   }
 
    retro_savestate_complete = false;
    retro_savestate_result = 0;
-- 
2.53.0

